library/sanitize.inc.php

```diff
@@ -55,5 +55,34 @@ function image_has_right_size($size) {
   return $size < 20971520;
 }
 
+// Generate csrf token for authentication
+function generateCsrfToken()
+{
+    if (!extension_loaded('openssl')) {
+      error_log("ERROR: openssl extension not enabled, needed for proper functioning of LibreHealth");
+        die("Error: Systems needs openssl.");
+    }
+
+    $csrfToken = bin2hex(openssl_random_pseudo_bytes(32));
+
+    if (empty($csrfToken)) {
+        error_log("ERROR : Token generation failed");
+        die("Error : Unable to correctly generate token");
+    }
+
+    return $csrfToken;
+}
+
+// Function to verify a csrf token
+function verifyCsrfToken($token)
+{
+    if (hash_equals($_SESSION['token'], $token)) {
+        return true;
+    } else {
```

Comment:
Great initiative @Ngai-E looking forward to this ... One thing can you create a class for this say something like `CsrfToken::class`

Reply:
Thanks, @muarachmann we will surely do that.